File: frontend/src/types/category.ts
export interface ResourceLink {
  title: string;
  url: string;
}

export interface LearningMaterial {
  title: string;
  url: string;
  type: string;
}

export interface Faq {
  question: string;
  answer: string;
}

export interface EscalationContact {
  role: string;
  when: string;
}

export interface Category {
  id: string;
  name: string;
  icon: string;
  description: string;
  defaultSituation: string;
  defaultDesiredOutcome: string;
  documentation: ResourceLink[];
  faqs: Faq[];
  learningMaterials: LearningMaterial[];
  escalation?: EscalationContact;
}

File: backend/ManagerCompass.Api/Models/Category.cs
namespace ManagerCompass.Api.Models;

public class ResourceLink
{
    public string Title { get; set; } = string.Empty;
    public string Url { get; set; } = string.Empty;
}

public class LearningMaterial
{
    public string Title { get; set; } = string.Empty;
    public string Url { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty; // e.g. "Course", "Video", "Guide"
}

public class Faq
{
    public string Question { get; set; } = string.Empty;
    public string Answer { get; set; } = string.Empty;
}

public class EscalationContact
{
    public string Role { get; set; } = string.Empty;
    public string When { get; set; } = string.Empty;
}

public class Category
{
    public string Id { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Icon { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string DefaultSituation { get; set; } = string.Empty;
    public string DefaultDesiredOutcome { get; set; } = string.Empty;
    public List<ResourceLink> Documentation { get; set; } = new();
    public List<Faq> Faqs { get; set; } = new();
    public List<LearningMaterial> LearningMaterials { get; set; } = new();
    public EscalationContact? Escalation { get; set; }
}
